site/src/components/Screens/CompleteSignupScreen.tsx

```diff
@@ -0,0 +1,152 @@
+import { useContext, useRef, useState, VFC } from "react";
+import { TextField } from "../TextField";
+import { Button } from "../Button";
+import { useShortnameTextField } from "../hooks/useShortnameTextField";
+import { apiClient } from "../../lib/apiClient";
+import {
+  ApiCompleteSignupRequestBody,
+  ApiCompleteSignupResponse,
+} from "../../pages/api/completeSignup.api";
+import UserContext from "../../context/UserContext";
+
+type CompleteSignupScreenProps = {
+  email: string;
+};
+
+export const CompleteSignupScreen: VFC<CompleteSignupScreenProps> = ({
+  email,
+}) => {
+  const { setUser } = useContext(UserContext);
+
+  const shortnameInputRef = useRef<HTMLInputElement>(null);
+  const preferredNameInputRef = useRef<HTMLInputElement>(null);
+
+  const [touchedPreferredNameInput, setTouchedPreferredNameInput] =
+    useState<boolean>(false);
+  const [touchedShortnameInput, setTouchedShortnameInput] =
+    useState<boolean>(false);
+
+  const [completingSignup, setCompletingSignup] = useState<boolean>(false);
+
+  const { shortname, setShortname, isShortnameValid, shortnameHelperText } =
+    useShortnameTextField({
+      touched: touchedShortnameInput,
+    });
+
+  const [preferredName, setPreferredName] = useState<string>("");
+
+  const isPreferredNameEmpty = !preferredName;
+
+  const isPreferredNameValid = !isPreferredNameEmpty;
+  const preferredNameHelperText = "";
+
+  const handleSubmit = async () => {
+    if (isShortnameValid && isPreferredNameValid) {
+      setCompletingSignup(true);
+      const { data, error } = await apiClient.post<
+        ApiCompleteSignupRequestBody,
+        ApiCompleteSignupResponse
+      >("completeSignup", { shortname, preferredName });
+      setCompletingSignup(false);
+
+      if (error) {
+        throw error;
+      } else if (data) {
+        setUser(data.user);
+      }
+    }
+  };
+
+  return (
+    <Box display="flex" flexDirection="column" alignItems="center">
+      <Icon
+        className="fa-solid fa-badge-check"
+        sx={{
+          fontSize: 50,
+          color: ({ palette }) => palette.purple[600],
+          marginBottom: 3,
+        }}
+      />
+      <Typography
+        variant="bpBodyCopy"
+        textAlign="center"
+        marginBottom={1}
+        sx={{
+          maxWidth: {
+            xs: "unset",
+            sm: "70%",
+          },
+        }}
+      >
+        Thanks for confirming your email <strong>{email}</strong>
+      </Typography>
+      <Typography
+        variant="bpHeading3"
+        textAlign="center"
+        marginBottom={2}
+        sx={{
+          color: ({ palette }) => palette.gray[80],
+          fontWeight: 500,
+        }}
+      >
+        Add your account details
+      </Typography>
+      <Box
+        width="100%"
+        component="form"
+        onSubmit={(event: React.FormEvent<HTMLFormElement>) => {
+          event.preventDefault();
+          void handleSubmit();
+        }}
```

Comment:
I'd move the `event.preventDefault()` to `handleSubmit` as the function is split at the moment where it doesn't really need to be(?)

Reply:
Yeah it doesn't need to be handled here, just like keeping the event of `handle` functions unless is needed. But will change as I don't have strong feelings on this.